<?xml version="1.0"?>
<launch>
    <arg name="rectify" default="false" />
    <arg name="name" default="arducam" />
    <node pkg="usb_cam" type="usb_cam_node" name="$(arg name)" required="true" output="screen">
        <param name="video_device" value="/dev/video0" />
        <param name="io_method" value="userptr" />
        <param name="pixel_format" value="yuyv" />
        <param name="camera_name" value="$(arg name)" />
        <param name="camera_frame_id" value="$(arg name)" />
        <param name="camera_info_url" value="package://dalek_apriltag/config/$(arg name)_calib.yaml" />
        <param name="image_width" value="1600" />
        <param name="image_height" value="1200" />
        <param name="framerate" value="5" />
        <rosparam command="load" file="$(find dalek_apriltag)/config/arducam_controls.yaml" />
    </node>
    <node if="$(arg rectify)" pkg="image_proc" type="image_proc" name="image_proc" ns="$(arg name)" required="true">
    </node>
</launch>
